packages/ava/src/data/utils/isType.ts

```diff
@@ -0,0 +1,83 @@
+import { isDateString } from './isDateString';
+
+export function isNil(value: unknown) {
+  return value === null || value === undefined || value === '' || Number.isNaN(value as number) || value === 'null';
+}
+
+export function isString(value: unknown) {
+  return typeof value === 'string';
+}
+
+export function isNumberString(value: string) {
+  let hasDot = false;
+  let tempValue = value;
+  if (/^[+-]/.test(tempValue)) {
+    tempValue = tempValue.slice(1);
+  }
+  for (let i = 0; i < tempValue.length; i += 1) {
+    const char = tempValue[i];
+    if (char === '.') {
+      if (hasDot === false) {
+        hasDot = true;
+      } else {
+        return false;
+      }
+    }
+    if (char !== '.' && !/[0-9]/.test(char)) {
+      return false;
+    }
+  }
+  return tempValue.trim() !== '';
+}
+
+export function isNumber(value: unknown, checkString?: boolean) {
+  if (typeof value === 'number') return true;
+  if (checkString && isString(value)) return isNumberString(value as string);
+  return false;
+}
+
+export function isInteger(value: unknown, checkString?: boolean) {
+  if (typeof value === 'number') return Number.isInteger(value);
+  if (checkString && isString(value) && isNumberString(value as string)) return !(value as string).includes('.');
+  return false;
+}
+
+export function isFloat(value: unknown, checkString?: boolean) {
+  if (typeof value === 'number') return !Number.isNaN(value) && !Number.isInteger(value);
+  if (checkString && isString(value) && isNumberString(value as string)) return (value as string).includes('.');
+  return false;
+}
+
+export function isDate(value: unknown, checkString?: boolean): boolean {
+  if (value && Object.getPrototypeOf(value) === Date.prototype) return true;
+  if (checkString && isString(value)) return isDateString(value as string);
+  return false;
+}
+
+export function isBoolean(value: unknown, checkSpecialBoolean?: boolean) {
+  return checkSpecialBoolean
+    ? [
+        [true, false],
+        [0, 1],
+        ['true', 'false'],
+        ['Yes', 'No'],
+        ['True', 'False'],
+        ['0', '1'],
+        ['是', '否'],
+      ].some((list: unknown[]) => {
+        return (value as unknown[]).every((item: unknown) => list.includes(item));
+      })
+    : typeof value === 'boolean';
+}
+
+export function isObject(value: unknown) {
+  return value && Object.getPrototypeOf(value) === Object.prototype;
```

Comment:
![image](https://user-images.githubusercontent.com/35586469/201505438-2085b0b7-4ecc-4857-834b-a3b61618e8e1.png)

This can be a problem when the value is null, see https://github.com/antvis/AVA/pull/477#discussion_r1018199260

Reply:
`data`'s object is the object we often talk about, it's different from the standard object. If it's not suitable to put it in the global, I can just use it here.

Reply:
ok